### PR TITLE
[MNG-7911] Require JDK 11 and support JDK 21 at build time

### PR DESCRIPTION
**Pull request by gnodet**
IT PR at https://github.com/apache/maven-integration-testing/pull/310

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
-        java: [8, 11, 17]
+        java: [11, 17, 21]
```
Comment (mthmulders) [inline .github/workflows/maven.yml:64]: There is no way to build on 8 **and** 21?

Comment (gnodet) [reply .github/workflows/maven.yml:64]: I must admit I haven't checked all the intermediate mockito versions, but the one currently in use does not work with JDK 21 and the most recent one is built with JDK 11 byte code.

Comment (mthmulders) [reply .github/workflows/maven.yml:64]: I can imagine the Mockito team 'made the jump' and dropped Java 8 altogether. This change would not _prevent_ running Maven on Java 8, would it?

Comment (gnodet) [reply .github/workflows/maven.yml:64]: Definitely not, it's only the build time requirements that are raised, but the target byte code is still JDK 8.

Comment (mthmulders) [reply .github/workflows/maven.yml:64]: Thanks for clarifying!

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ jobs:
 
       - uses: actions/setup-java@v3
         with:
-          java-version: 8
+          java-version: 11
```
Comment (mthmulders) [inline .github/workflows/maven.yml:45]: Should we also update the Jenkinsfile?

File: maven-model-builder/src/test/java/org/apache/maven/model/building/BuildModelSourceTransformerTest.java [hunk 3]
```diff
@@ -136,6 +137,10 @@ public static boolean equalsDeep(Object m1, Object m2) {
             }
             BeanInfo bean = Introspector.getBeanInfo(m1.getClass());
             for (PropertyDescriptor prop : bean.getPropertyDescriptors()) {
+                if (("first".equals(prop.getName()) || "last".equals(prop.getName()))
```
Comment (mthmulders) [inline maven-model-builder/src/test/java/org/apache/maven/model/building/BuildModelSourceTransformerTest.java:140]: How is this related to building on JDK 21?

File: pom.xml [hunk 4]
```diff
@@ -642,6 +679,26 @@ under the License.
               <fail>true</fail>
             </configuration>
           </execution>
+          <execution>
+            <id>enforce-bytecode-version</id>
```
Comment (mthmulders) [inline pom.xml:683]:
```suggestion
            <id>enforce-build-jdk-version</id>
```

File: maven-model-builder/src/test/java/org/apache/maven/model/building/BuildModelSourceTransformerTest.java (hunk 3 above)
Comment (rmannibucau) [reply maven-model-builder/src/test/java/org/apache/maven/model/building/BuildModelSourceTransformerTest.java:140]: New sequence api i guess

Comment (gnodet) [reply maven-model-builder/src/test/java/org/apache/maven/model/building/BuildModelSourceTransformerTest.java:140]: Exactly, see https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/SequencedCollection.html

Those getters are throwing `NoSuchElementException` when the list is empty.

File: pom.xml (hunk 4 above)
Comment (gnodet) [reply pom.xml:683]: This execution overrides the one defined in the parent pom, with the difference that it adds the `ignoredScopes` configuration bit.  So it cannot be renamed...

Comment (mthmulders) [reply pom.xml:683]: Ah, wasn't aware of that. Makes sense then!

**Review by mthmulders — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

File: .github/workflows/maven.yml (hunk 2 above)
Comment (gnodet) [reply .github/workflows/maven.yml:45]: I update the Jenkins file to build with JDK 11, 17 and 21.

**Review by mthmulders — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #9002
